packages/editor/src/functions/EditorControlFunctions.test.tsx

```diff
@@ -0,0 +1,855 @@
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import assert from 'assert'
+
+import { EntityUUID } from '@etherealengine/common/src/interfaces/EntityUUID'
+import { EventDispatcher } from '@etherealengine/engine/src/common/classes/EventDispatcher'
+import { destroyEngine, Engine } from '@etherealengine/engine/src/ecs/classes/Engine'
+import { EngineState } from '@etherealengine/engine/src/ecs/classes/EngineState'
+import { Entity } from '@etherealengine/engine/src/ecs/classes/Entity'
+import { SceneSnapshotSystem, SceneState } from '@etherealengine/engine/src/ecs/classes/Scene'
+import { getComponent, hasComponent } from '@etherealengine/engine/src/ecs/functions/ComponentFunctions'
+import { EntityTreeComponent } from '@etherealengine/engine/src/ecs/functions/EntityTree'
+import { SystemDefinitions } from '@etherealengine/engine/src/ecs/functions/SystemFunctions'
+import { createEngine } from '@etherealengine/engine/src/initializeEngine'
+import { PhysicsState } from '@etherealengine/engine/src/physics/state/PhysicsState'
+import { NameComponent } from '@etherealengine/engine/src/scene/components/NameComponent'
+import { ShadowComponent } from '@etherealengine/engine/src/scene/components/ShadowComponent'
+import { UUIDComponent } from '@etherealengine/engine/src/scene/components/UUIDComponent'
+import { FogType } from '@etherealengine/engine/src/scene/constants/FogType'
+import { SceneLoadingSystem } from '@etherealengine/engine/src/scene/SceneModule'
+import { SceneDataType, SceneID, SceneJsonType } from '@etherealengine/engine/src/schemas/projects/scene.schema'
+import { UserID } from '@etherealengine/engine/src/schemas/user/user.schema'
+import { LocalTransformComponent } from '@etherealengine/engine/src/transform/components/TransformComponent'
+import { applyIncomingActions, getMutableState } from '@etherealengine/hyperflux'
+import { act, render } from '@testing-library/react'
+import React from 'react'
+import { FogSettingsComponent } from '../../../engine/src/scene/components/FogSettingsComponent'
+import { EditorControlFunctions } from './EditorControlFunctions'
+
+const testScene = {
```

Comment:
let's put this in a .json file and import it to improve readability of this file